google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/models/ConditionalRowMutation.java

```diff
@@ -146,9 +170,28 @@ public CheckAndMutateRowRequest toProto(RequestContext requestContext) {
    */
   @BetaApi
   public static ConditionalRowMutation fromProto(@Nonnull CheckAndMutateRowRequest request) {
-    String tableId = NameUtil.extractTableIdFromTableName(request.getTableName());
-    ConditionalRowMutation rowMutation =
-        ConditionalRowMutation.create(tableId, request.getRowKey());
+    String tableName = request.getTableName();
+    String authorizedViewName = request.getAuthorizedViewName();
+
+    Preconditions.checkArgument(
+        !tableName.isEmpty() || !authorizedViewName.isEmpty(),
+        "Either table name or authorized view name must be specified");
+    Preconditions.checkArgument(
+        tableName.isEmpty() || authorizedViewName.isEmpty(),
+        "Table name and authorized view name cannot be specified at the same time");
+
```

Comment:
here and everywhere can you extract the repeated logic to a helper in NameUtil? ie
```java
static TargetId extractTargetId(@Nullable tableName, @Nullable authViewName) {
  Preconditions.checkArgument(...)
  if (!tableName.isEmpty()) {
     ...
     return TableId.of(...)
  }
  ...
}